TodoListSPA/app.js

```diff
@@ -0,0 +1,238 @@
+ * acquire token with one of the UI-ed functions acquireTokenPopup or acquireTokenRedirect
+ */
+function acquireAnAccessTokenAndCallTheProtectedService() {
+    showProgress("acquiring an access token for the Web API");
+    authenticationContext.acquireToken(webApiConfig.resourceId, function (errorDesc, token, error) {
+        if (error) {
+            if (config.popUp) {
+                authenticationContext.acquireTokenPopup(webApiConfig.resourceId, null, null, onAccessToken);
+            }
+            else {
+                authenticationContext.acquireTokenRedirect(webApiConfig.resourceId, null, null);
+            }
+        }
+        else {
+            onAccessToken(errorDesc, token, error);
+        }
+    });
+}
+
+/**
+ * Function called when the access token is available. This will call the service
+ * @param {string} errorDesc - Error message
+ * @param {string} token - Access token to use in the Web API
+ * @param {string} error - Error code from the STS
+ */
+function onAccessToken(errorDesc, token, error) {
+    if (error) {
+        showError("acquireToken", error);
+    }
+    if (token) {
+        showProgress("calling the Web API with the access token");
+ //       callServiceWithToken(token, webApiConfig.resourceBaseAddress + "api/todolist");
+        callServiceWithToken(token, webApiConfig.resourceBaseAddress + "api/AccessCaApi");
+    }
+}
+
+
+/**
+ * Calls the service with the access token. Also handles the "claim challenge", that is when the service requires
+ * more claims from the users (for instance that the users does two factors authentication)
+ * @param {string} token - Access token for the web API
+ * @param {string} endpoint - endpoint to the Web API to call
+ */
+function callServiceWithToken(token, endpoint) {
+    // Header won't work in IE, but you could replace it with a call to AJAX JQuery for instance.
+    var headers = new Headers();
+    var bearer = "Bearer " + token;
+    headers.append("Authorization", bearer);
+    var options = {
+        method: "GET",
+        headers: headers
+    };
+
+    // Note that fetch API is not available in all browsers
+    fetch(endpoint, options)
+        .then(function (response) {
+            var contentType = response.headers.get("content-type");
+
+            if (response.status === 200 && contentType && contentType.indexOf("application/json") !== -1) {
+                // Case where we got the content from the API (as JSon)
+                response.json()
+                    .then(function (data) {
+                        // Display response in the page
+                        showProgress("");
+                        showAPIResponse(data, token);
+                    })
+                    .catch(function (error) {
+                        showError(endpoint, error);
+                    });
+            } else if (response.status === 403 && contentType && contentType.indexOf("text/plain; charset=utf-8") !== -1) {
```

Comment:
This looks good. One thing you could consider is adding an optional parameter to your AT handler function for claims. This way you don't do acquiretoken calls in various places. Not a huge deal.